main/midi/midiPlayer.h

```diff
@@ -0,0 +1,722 @@
+    CLOSED_HI_HAT  = 42,
+    HIGH_FLOOR_TOM = 43,
+    /// @brief This note supersedes any ::CLOSED_HI_HAT, ::PEDAL_HI_HAT, or ::OPEN_HI_HAT notes playing
+    PEDAL_HI_HAT = 44,
+    LOW_TOM      = 45,
+    /// @brief This note supersedes any ::CLOSED_HI_HAT, ::PEDAL_HI_HAT, or ::OPEN_HI_HAT notes playing
+    OPEN_HI_HAT     = 46,
+    LOW_MID_TOM     = 47,
+    HIGH_MID_TOM    = 48,
+    CRASH_CYMBAL_1  = 49,
+    HIGH_TOM        = 50,
+    RIDE_CYMBAL_1   = 51,
+    CHINESE_CYMBAL  = 52,
+    RIDE_BELL       = 53,
+    TAMBOURINE      = 54,
+    SPLASH_CYMBAL   = 55,
+    COWBELL         = 56,
+    CRASH_CYMBAL_2  = 57,
+    VIBRASLAP       = 58,
+    RIDE_CYMBAL_2   = 59,
+    HIGH_BONGO      = 60,
+    LOW_BONGO       = 61,
+    MUTE_HIGH_CONGA = 62,
+    OPEN_HIGH_CONGA = 63,
+    LOW_CONGA       = 64,
+    HIGH_TIMBALE    = 65,
+    LOW_TIMBALE     = 66,
+    HIGH_AGOGO      = 67,
+    LOW_AGOGO       = 68,
+    CABASA          = 69,
+    MARACAS         = 70,
+    /// @brief This note supersdes any ::SHORT_WHISTLE or ::LONG_WHISTLE notes playing
+    SHORT_WHISTLE = 71,
+    /// @brief This note supersdes any ::SHORT_WHISTLE or ::LONG_WHISTLE notes playing
+    LONG_WHISTLE = 72,
+    /// @brief This note supersdes any ::SHORT_GUIRO or ::LONG_GUIRO notes playing
+    SHORT_GUIRO = 73,
+    /// @brief This note supersdes any ::SHORT_GUIRO or ::LONG_GUIRO notes playing
+    LONG_GUIRO     = 74,
+    CLAVES         = 75,
+    HIGH_WOODBLOCK = 76,
+    LOW_WOODBLOCK  = 77,
+    /// @brief This note supersdes any ::SHORT_GUIRO or ::LONG_GUIRO notes playing
+    MUTE_CUICA = 78,
+    /// @brief This note supersdes any ::SHORT_GUIRO or ::LONG_GUIRO notes playing
+    OPEN_CUICA = 79,
+    /// @brief This note supersdes any ::MUTE_TRIANGLE or ::OPEN_TRIANGLE notes playing
+    MUTE_TRIANGLE = 80,
+    /// @brief This note supersdes any ::MUTE_TRIANGLE or ::OPEN_TRIANGLE notes playing
+    OPEN_TRIANGLE = 81,
+    // Roland GS Extensions
+    /*SHAKER = 82,
+    JINGLE_BELL = 83,
+    BELLTREE = 84,
+    CASTANETS = 85,
+    MUTE_SURDO = 86,
+    OPEN_SURDO = 87,*/
+    // End Roland GS Extensions
+} percussionNote_t;
+
+//==============================================================================
+// Structs
+//==============================================================================
+
+/**
+ * @brief Describes the characteristics of a particular timbre while
+ *
+ */
+typedef struct
+{
+    // Time taken to ramp up to full volume
+    int32_t attack;
+
+    // Time taken for the volume to fade to the sustain volume
+    int32_t decay;
+
+    // Time it takes to silence the note after release
+    int32_t release;
+
+    // The volume of the sustain note, proportional to the original volume
+    uint8_t sustain;
+
+    /// @brief The index of the repeat loop
+    // uint32_t loopStart;
+    /// @brief The index at which the loop should repeat
+    // uint32_t loopEnd;
+
+} envelope_t;
```

Comment:
You went all out! I was perfectly happy to pretend ADSR didn't exist

Reply:
Oh, heh, this doesn't actually do anything yet 😅 probably should delete it for now. Though I think I will come back and try to add support for it pretty soon anyway, it shouldn't be too complicated.

Reply:
Nah, leave it with a `TODO`. Its not a good project until it's littered with those.